### DISCUSSION TITLE
ESP32-S3 The operation can’t be completed because one or more required items can’t be found. (Error code -43)

### DISCUSSION THREAD

**Discussion by marchingband** [adafruit/tinyuf2#392] [Q&A]
I am using adafruit_qtpy_esp32s3 on a custom PCB with ESP32S3, and the uf2 bootloader works.
Sometimes, when drag-and-dropping the new firmware, on macos (Mac M1), I get this error popup from the OS:
"The operation can’t be completed because one or more required items can’t be found. (Error code -43)"
The new firmware does boot, which is odd, but I am wondering what this error means, and how I can avoid it.
This is for a product and I don't want my users to encounter this error, if possible.
Thanks!

**Comment by VirusPilot** [comment]
Same experience here but I am afraid this cannot be easily resolved as it would require some sort of "safe eject" on your Mac, initiated by the esp32s3 module.

**Comment by marchingband** [comment]
@VirusPilot  Hmm. It's odd because some .uf2 files cause it, others do not, each consistently. I have not been able to determine the difference between the files that would cause it.

**Reply by VirusPilot** [reply]
That is indeed strange.

**Comment by dhalbert** [comment]
Which versions of macOS are you all using?

We have had other reports of this. macOS has been churning the FAT implementation for a while now, changing it from a kernel-based to a user-spaced implementation, and not for the better. macOS 14.1 had severe delays completing writes to small filesystems, which caused write errors and incomplete writes. macOS 14.4 fixed the delays but started doing many unnecessary writes, causing writes to filesystems <= 1GB to be about 40 times slower than writes to larger filesystems.

A simple delay after the UF2 write completes, and before the restart, _might_ fix the problem described above.

**Comment by VirusPilot** [comment]
My iMac has 14.5 (23F79) running. I have a separate issue formatting a T-Beam partition as FAT, had to do it on a Windows PC.

**Comment by marchingband** [comment]
Im on Ventura, 13.4
